File: caesar.c
#include <cs50.h>
#include<stdio.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Error in given input!!\n");
        return 1;
    }
    else
    {
        int c;
        int k = atoi(argv[1]);
        
        if(k<0)
        {
            printf("enter a positive number");
            return 0;
        }
        printf("plaintext: ");
        string plaintext=get_string();
        printf("ciphertext: ");
        
        for(int i = 0, n = strlen(plaintext); i < n; i++)
        {
            if (isalpha(plaintext[i]) && isupper(plaintext[i]))
            {
                c = (plaintext[i] - 65 + k) % 26;
                printf("%c", c + 65);
            }
            else if (isalpha(plaintext[i]) && islower(plaintext[i]))
            {
                c = (plaintext[i] - 97 + k) % 26;
                printf("%c", c + 97);
            }
            else
                printf("%c",plaintext[i]);
        }
        printf("\n");
        return 0;
    }
}

File: caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("You should provide cmd line arguments!")
        exit(1)
    
    key = int(sys.argv[1])
    print("enter the string")
    plain = cs50.get_string()
    cipher = []
    for i in plain:
        if i.isalpha():
             cipher.append(caesar(i, key))     
        else:
            cipher.append(i)
                
    print("".join(cipher))
    exit(0)

def caesar(c, key):
    if c.islower():
        return chr(((ord(c) - 97 + key) % 26) + 97)
    else:
        return chr(((ord(c) - 65 + key) % 26) + 65)
        
if __name__ == "__main__":
    main()